src/utils/hooks/use-validation-schema.ts

```diff
@@ -1,20 +1,82 @@
 import { useContext, useEffect, useState } from "react";
 import * as Yup from "yup";
 import { ObjectShape } from "yup/lib/object";
-import { YupContext, YupHelper } from "../../components/frontend-engine/yup";
+import { TFrontendEngineValues } from "../../components/frontend-engine/types";
+import { IYupValidationRule, YupContext, YupHelper } from "../../components/frontend-engine/yup";
 
 /**
  * Hook that handles the generation of the validationSchema
  */
 export const useValidationSchema = () => {
```

Comment:
there are some changes to the `useValidationSchema` hook in the `publish` branch:
- `formValidationConfig` and `setFormValidationConfig` are removed from this hook
- hook only returns validation schema
- hook does not return anything to do with validation config